buildSrc/src/main/kotlin/datadog/gradle/plugin/testJvmConstraints/TestJvmSpec.kt

```diff
@@ -1,88 +1,162 @@
 package datadog.gradle.plugin.testJvmConstraints
 
-import org.gradle.kotlin.dsl.support.serviceOf
 import org.gradle.api.GradleException
 import org.gradle.api.Project
+import org.gradle.api.internal.provider.PropertyFactory
+import org.gradle.api.provider.Provider
+import org.gradle.jvm.toolchain.JavaLanguageVersion
 import org.gradle.jvm.toolchain.JavaLauncher
 import org.gradle.jvm.toolchain.JavaToolchainService
+import org.gradle.jvm.toolchain.JavaToolchainSpec
+import org.gradle.jvm.toolchain.JvmImplementation
+import org.gradle.jvm.toolchain.JvmVendorSpec
+import org.gradle.jvm.toolchain.internal.DefaultToolchainSpec
+import org.gradle.jvm.toolchain.internal.SpecificInstallationToolchainSpec
+import org.gradle.kotlin.dsl.support.serviceOf
 import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.Paths
 
-
+/**
+ * Handles the `testJvm` property to resolve a Java launcher for testing.
+ *
+ * The `testJvm` property can be set via command line or environment variable to specify
+ * which JVM to use for running tests. E.g.
+ *
+ * ```shell
+ * ./gradlew test -DtestJvm=ZULU11
```

Comment:
```suggestion
 * ./gradlew test -PtestJvm=ZULU11
```

I think we should still be using `-P` here, since on line 44 below (`project.providers.gradleProperty(TEST_JVM)`), we are using `testJvm` as a Gradle project property ([ref](https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties))